### DISCUSSION TITLE
alpinejs only for small project, why would have this statement existing

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3975] [4. General]
Hi,

I am not sure how to measure a project is big or small, but google has voice alpinejs is only for lightweight projects, and how to measure small or big, or that statement is correct?

Thanks
Eric Xin

**Comment by ekwoka** [comment]
The statement is not correct.

AlpineJS is for providing a Reactive UI to HTML that isn't rendered by JavaScript, like that rendered by PHP, C#, Go, Rust, Python, etc.

It is not, on it's own, remotely competitive against full UI rendering frameworks like Vue or SvelteKit. It's not trying to be. It serves an entirely different concept.

**Reply by ericxin1982** [reply]
I see, that is two different way, no wonder Alpine JS and htmx would compare or use together, since hypermedia this terms.

**Reply by ekwoka** [reply]
They work well together, aside from some strange nastiness that happens on the edge.

**Reply by ericxin1982** [reply]
Thanks.
